File: api/providers/data_provider.py
from models.warehouses import Warehouses
from models.locations import Locations
from models.transfers import Transfers
from models.items import Items
from models.item_lines import ItemLines
from models.item_groups import ItemGroups
from models.item_types import ItemTypes
from models.inventories import Inventories
from models.suppliers import Suppliers
from models.orders import Orders
from models.clients import Clients
from models.shipments import Shipments
import os


DEBUG = False

ROOT_PATH = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))), 'data/')
ROOT_PATH = ROOT_PATH.replace('\\', '/') 

_warehouses = None
_locations = None
_transfers = None
_items = None
_item_lines = None
_item_groups = None
_item_types = None
_inventories = None
_suppliers = None
_orders = None
_shipments = None
_clients = None


def init():
    global _warehouses
    _warehouses = Warehouses(ROOT_PATH, DEBUG)
    global _locations
    _locations = Locations(ROOT_PATH, DEBUG)
    global _transfers
    _transfers = Transfers(ROOT_PATH, DEBUG)
    global _items
    _items = Items(ROOT_PATH, DEBUG)
    global _item_lines
    _item_lines = ItemLines(ROOT_PATH, DEBUG)
    global _item_groups
    _item_groups = ItemGroups(ROOT_PATH, DEBUG)
    global _item_types
    _item_types = ItemTypes(ROOT_PATH, DEBUG)
    global _inventories
    _inventories = Inventories(ROOT_PATH, DEBUG)
    global _suppliers
    _suppliers = Suppliers(ROOT_PATH, DEBUG)
    global _orders
    _orders = Orders(ROOT_PATH, DEBUG)
    global _clients
    _clients = Clients(ROOT_PATH, DEBUG)
    global _shipments
    _shipments = Shipments(ROOT_PATH, DEBUG)


def fetch_warehouse_pool():
    return _warehouses


def fetch_location_pool():
    return _locations


def fetch_transfer_pool():
    return _transfers


def fetch_item_pool():
    return _items


def fetch_item_line_pool():
    return _item_lines


def fetch_item_group_pool():
    return _item_groups


def fetch_item_type_pool():
    return _item_types


def fetch_inventory_pool():
    return _inventories


def fetch_supplier_pool():
    return _suppliers


def fetch_order_pool():
    return _orders


def fetch_client_pool():
    return _clients


def fetch_shipment_pool():
    return _shipments

File: C#api/providers/data_provider.cs
using System;
using Models;
using System.IO;

class DataProvider
{
    private static readonly bool DEBUG = false;
    private static readonly string ROOT_PATH = Path.Combine(Directory.GetParent(Directory.GetCurrentDirectory()).Parent.Parent.FullName, "data").Replace('\\', '/');

    private static Warehouses _warehouses;
    private static Locations _locations;
    private static Transfers _transfers;
    private static Items _items;
    private static ItemLines _itemLines;
    private static ItemGroups _itemGroups;
    private static ItemTypes _itemTypes;
    private static Inventories _inventories;
    private static Suppliers _suppliers;
    private static Orders _orders;
    private static Clients _clients;
    private static Shipments _shipments;


    public static void Init()
    {
        _warehouses = new Warehouses(ROOT_PATH, DEBUG);
        _locations = new Locations(ROOT_PATH, DEBUG);
        _transfers = new Transfers(ROOT_PATH, DEBUG);
        _items = new Items(ROOT_PATH, DEBUG);
        _itemLines = new ItemLines(ROOT_PATH, DEBUG);
        _itemGroups = new ItemGroups(ROOT_PATH, DEBUG);
        _itemTypes = new ItemTypes(ROOT_PATH, DEBUG);
        _inventories = new Inventories(ROOT_PATH, DEBUG);
        _suppliers = new Suppliers(ROOT_PATH, DEBUG);
        _orders = new Orders(ROOT_PATH, DEBUG);
        _clients = new Clients(ROOT_PATH, DEBUG);
        _shipments = new Shipments(ROOT_PATH, DEBUG);
    }

    public static Warehouses fetch_warehouse_pool() => _warehouses;
    public static Locations fetch_location_pool() => _locations;
    public static Transfers fetch_transfer_pool() => _transfers;
    public static Items fetch_item_pool() => _items;
    public static ItemLines fetch_itemline_pool() => _itemLines;
    public static ItemGroups fetch_itemgroup_pool() => _itemGroups;
    public static ItemTypes fetch_itemtype_pool() => _itemTypes;
    public static Inventories fetch_inventory_pool() => _inventories;
    public static Suppliers fetch_supplier_pool() => _suppliers;
    public static Orders fetch_order_pool() => _orders;
    public static Clients fetch_client_pool() => _clients;
    public static Shipments fetch_shipment_pool() => _shipments;
}
